wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -0,0 +1,1599 @@
+            }
+
+            String branch = fullBranch.substring(Constants.R_HEADS.length());
+
+            StoredConfig config = repository.getConfig();
+            if (remoteName == null) {
+                remoteName = config.getString(ConfigConstants.CONFIG_BRANCH_SECTION, branch,
+                                              ConfigConstants.CONFIG_KEY_REMOTE);
+                if (remoteName == null) {
+                    remoteName = Constants.DEFAULT_REMOTE_NAME;
+                }
+            }
+            remoteUri = config.getString(ConfigConstants.CONFIG_REMOTE_SECTION, remoteName, ConfigConstants.CONFIG_KEY_URL);
+
+            String remoteBranch;
+            RefSpec fetchRefSpecs = null;
+            String refSpec = request.getRefSpec();
+            if (refSpec != null) {
+                fetchRefSpecs = (refSpec.indexOf(':') < 0) //
+                                ? new RefSpec(Constants.R_HEADS + refSpec + ":" + fullBranch) //
+                                : new RefSpec(refSpec);
+                remoteBranch = fetchRefSpecs.getSource();
+            } else {
+                remoteBranch = config.getString(ConfigConstants.CONFIG_BRANCH_SECTION, branch,
+                                                ConfigConstants.CONFIG_KEY_MERGE);
+            }
+
+            if (remoteBranch == null) {
+                remoteBranch = fullBranch;
+            }
+
+            FetchCommand fetchCommand = getGit().fetch();
+            fetchCommand.setRemote(remoteName);
+            if (fetchRefSpecs != null) {
+                fetchCommand.setRefSpecs(fetchRefSpecs);
+            }
+            int timeout = request.getTimeout();
+            if (timeout > 0) {
+                fetchCommand.setTimeout(timeout);
+            }
+
+            FetchResult fetchResult = (FetchResult)executeRemoteCommand(remoteUri, fetchCommand);
+
+            Ref remoteBranchRef = fetchResult.getAdvertisedRef(remoteBranch);
+            if (remoteBranchRef == null) {
+                remoteBranchRef = fetchResult.getAdvertisedRef(Constants.R_HEADS + remoteBranch);
+            }
+            if (remoteBranchRef == null) {
+                throw new GitException(String.format(ERROR_PULL_REF_MISSING, remoteBranch));
+            }
+            org.eclipse.jgit.api.MergeResult mergeResult = getGit().merge().include(remoteBranchRef).call();
+            if (mergeResult.getMergeStatus().equals(org.eclipse.jgit.api.MergeResult.MergeStatus.ALREADY_UP_TO_DATE)) {
+                return newDto(PullResponse.class).withCommandOutput("Already up-to-date");
+            }
+
+            if (mergeResult.getConflicts() != null) {
+                StringBuilder message = new StringBuilder(ERROR_PULL_MERGE_CONFLICT_IN_FILES);
+                message.append("\n");
+                Map<String, int[][]> allConflicts = mergeResult.getConflicts();
+                for (String path : allConflicts.keySet()) {
+                    message.append(path).append("\n");
```

Comment:
is it http://docs.oracle.com/javase/8/docs/api/java/lang/System.html#lineSeparator-- ?